test/integration.test.ts

```diff
@@ -0,0 +1,176 @@
+        from: u,
+        chat: noChat ? undefined : c,
+    } as Message;
+    const update = {
+        message: m,
+    } as Update;
+
+    return update;
+};
+
+describe("Integration", () => {
+    describe("setCommands", () => {
+        it("should call setMyCommands with valid commands", async () => {
+            const myCommands = new Commands();
+            myCommands.command("command", "_", (_, next) => next());
+
+            const setMyCommandsSpy = spy(resolvesNext([true] as const));
+
+            await myCommands.setCommands({
+                api: {
+                    raw: { setMyCommands: setMyCommandsSpy },
+                } as unknown as Api,
+            });
+
+            assertSpyCalls(setMyCommandsSpy, 1);
+            assertSpyCall(setMyCommandsSpy, 0, {
+                args: [{
+                    commands: [{ command: "command", description: "_" }],
+                    language_code: undefined,
+                    scope: {
+                        type: "default",
+                    },
+                }],
+            });
+        });
+
+        it("should error when commands have custom prefixes", async () => {
+            const myCommands = new Commands({ prefix: "!" });
+            myCommands.command("command", "_", (_, next) => next());
+
+            const setMyCommandsSpy = spy(resolvesNext([true] as const));
+
+            await assertRejects(() =>
+                myCommands.setCommands({
+                    api: {
+                        raw: { setMyCommands: setMyCommandsSpy },
+                    } as unknown as Api,
+                })
+            );
+
+            assertSpyCalls(setMyCommandsSpy, 0);
+        });
+    });
+
+    describe("ctx.setMyCommands", () => {
+        it("should call setMyCommands with valid commands", async () => {
+            const myCommands = new Commands();
+            myCommands.command("command", "_", (_, next) => next());
+
+            const setMyCommandsSpy = spy(resolvesNext([true] as const));
+            const bot = new Bot<Context & CommandsFlavor>("dummy_token", {
+                botInfo: {
+                    id: 1,
+                    is_bot: true,
+                    username: "",
+                    can_join_groups: true,
+                    can_read_all_group_messages: true,
+                    supports_inline_queries: true,
+                    first_name: "",
+                },
+            });
+
+            bot.api.config.use(async (prev, method, payload, signal) => {
+                if (method !== "setMyCommands") {
+                    return prev(method, payload, signal);
+                }
+                await setMyCommandsSpy(payload);
+
+                return { ok: true, result: true as ReturnType<typeof prev> };
+            });
+
+            bot.use(commands());
+
+            bot.use(async (ctx, next) => {
+                await ctx.setMyCommands(myCommands);
+                await next();
+            });
+
```

Comment:
Missing test's for top level command setting bot call:
```suggestion
            await myCommands.setCommands(bot)
```

Reply:
I don't see why we need this here
This is tested on the previous `describe` block